fluidefi/fluidefi-caspernet-aggregator-tools
# Deploy consumer

**Summary of change:**
- Added a new micro-service, deploy consumer, which uses deployFetcher, deployParser, and deploySaver to consume a deploy. 

**List of tasks:** 

- [x] create mocks (MockDeployFetcher, MockDeployParser, MockDeploySaver)
- [x] create deploy fetcher unit test.
- [x] create deploy consumer unit test.
- [x] create deploy consumer integration test
- [x] update user guide
- [x] usage test: Last check for the performance of the service where the developer imitates the user experience by importing the services in a similar way to the npm package.

**Steps for testing**
- Make sure your local git is up to date
```bash
git pull
```

- Checkout the branch `feature/deploy-consumer`
```bash
git checkout feature/deploy-consumer
```

- Run the test across the test-net
```bash
npm run test
```

- Confirm deploy-consumer test has passed.
- Run the docs using live server and make sure it have been updated. 

**Proof of Success**

![Screenshot 2022-12-11 at 9 24 03 PM](https://user-images.githubusercontent.com/99526005/206918860-8ff87234-412e-4912-b8a9-94daf191352b.png)
